update ToolTipNoPermBadge component

flyinghermit opened the pull request:
- update so that it works for both "missing permission" and "missing enterprise license"
- adds `sticky` and `tooltipText` props, `tooltipText` types
- updated tests

Update for https://github.com/gravitational/teleport.e/pull/2052

flyinghermit committed:
update ToolTipNoPermBadg:
- works for both missing permission and missing enterprise license
- sticky and tooltipText props, tooltipText types
- updated tests

flyinghermit committed:
Merge branch 'master' into sshah/update-tooltip-no-perm-badge

ibeckermayer commented on web/packages/teleport/src/components/ToolTipNoPermBadge/ToolTipNoPermBadge.tsx:
```diff
@@ -71,16 +77,27 @@ export const ToolTipNoPermBadge: React.FC<Props> = ({
+          px={3}
+          py={2}
+          data-testid="tooltip-msg"
+          modalCss={() => `pointer-events: none`}
```
What is `modalCss`? AFAICT `modalCss` is not doing anything in this component. (Though it works in `Popover` because [`Popover` passes it through to a `Modal`](https://github.com/gravitational/teleport/blob/d1664b7f39742a2c2c27b244d0abe9a58878aa06/web/packages/design/src/Popover/Popover.jsx#L323))

sshahcodes replied:
You are right. Removed in https://github.com/gravitational/teleport/pull/31022/commits/3a71196fe14e220df8686c0085045d1ed19cd07a

ibeckermayer reviewed (commented):
I don't quite understand the behavior just from looking at the code, can you perhaps add a couple of screenshots?

ibeckermayer commented:
> I don't quite understand the behavior just from looking at the code, can you perhaps add a couple of screenshots?

Scratch that, I see you have screenshots in https://github.com/gravitational/teleport.e/pull/2052

kimlisa commented on web/packages/teleport/src/components/ToolTipNoPermBadge/ToolTipNoPermBadge.tsx:
```diff
@@ -71,16 +77,27 @@ export const ToolTipNoPermBadge: React.FC<Props> = ({
   max-width: 350px;
 `;
+
+export enum TooltipText {
```
```suggestion
export enum BadgeTitle {
```

kimlisa commented on web/packages/teleport/src/components/ToolTipNoPermBadge/ToolTipNoPermBadge.tsx:
```diff
@@ -16,15 +16,19 @@
 
 type Props = {
   borderRadius?: number;
+  tooltipText?: TooltipText;
```
tooltip text sounded like the text for popover
```suggestion
  badgeTitle?: BadgeTitle;
```

sshahcodes replied:
Updated this and below comments in https://github.com/gravitational/teleport/pull/31022/commits/3a71196fe14e220df8686c0085045d1ed19cd07a

kimlisa commented on web/packages/teleport/src/components/ToolTipNoPermBadge/ToolTipNoPermBadge.tsx:
```diff
@@ -16,15 +16,19 @@
 export const ToolTipNoPermBadge: React.FC<Props> = ({
   children,
   borderRadius = 2,
+  tooltipText,
```
```suggestion
  badgeTitle = BadgeTitle.LackingPermissions,
```

kimlisa commented on web/packages/teleport/src/components/ToolTipNoPermBadge/ToolTipNoPermBadge.tsx:
```diff
@@ -37,13 +41,15 @@ export const ToolTipNoPermBadge: React.FC<Props> = ({
   }
 
+  const popoverPointerEvent = !sticky ? 'none' : 'auto';
+  const tooltip = tooltipText ? tooltipText : TooltipText.LackingPermissions;
```
since we set a default in line 30, we prolly don't need line 45

kimlisa commented on web/packages/teleport/src/components/ToolTipNoPermBadge/ToolTipNoPermBadge.tsx:
```diff
@@ -55,10 +61,10 @@ export const ToolTipNoPermBadge: React.FC<Props> = ({
       </Box>
       <Popover
-        modalCss={() => `pointer-events: none;`}
+        modalCss={() => `pointer-events: ${popoverPointerEvent}`}
```
nit: simple enough for inline
```suggestion
        modalCss={() => `pointer-events: ${sticky ? 'auto' : 'none'}`}
```

flyinghermit committed:
address review comments:
- rename toolTipText to badgeTitle
- default badgeTitle BadgeTitle.LackingPermissions
- remove modalCss from StyledOnHover

flyinghermit committed:
update story for badgeTitle and sticky props

flyinghermit committed:
run prettier

sshahcodes commented:
Friendly ping for review please @ryanclark @ibeckermayer

ryanclark commented on web/packages/teleport/src/components/ToolTipNoPermBadge/ToolTipNoPermBadge.story.tsx:
```diff
@@ -40,6 +40,28 @@ export const NoPermissionBadgeComp = () => (
+export const StickyTooltip = () => (
+  <SomeBox>
+    I'm a sample container
+    <ToolTipNoPermBadge
+      sticky={true}
+      children={
+        <Box p={3}>I will only disappear after this component is unhovered</Box>
+      }
+    />
```
Any reason for the `children` prop over this?

```suggestion
    <ToolTipNoPermBadge
      sticky={true}
    >
        <Box p={3}>I will only disappear after this component is unhovered</Box>
    </ToolTipNoPermBadge>    
```

sshahcodes replied:
Nothing in particular, I actually started with just that but later found other stories(on the same page for `ToolTipNoPermBadge` component) using `children`. So for consistency, I used it here as well.

ryanclark reviewed (approved):
Sorry I was out on PTO

sshahcodes commented:
> Sorry I was out on PTO

Thanks for the review Ryan, and sorry for PTO ping. I only noticed it later.
